enaml/qt/qt_mpl_canvas.py

```diff
@@ -73,27 +76,48 @@ def _refresh_mpl_widget(self):
         """ Create the mpl widget and update the underlying control.
 
         """
-        # Delete the old widgets in the layout, it's just shenanigans
-        # to try to reuse the old widgets when the figure changes.
         widget = self.widget
         layout = widget.layout()
-        while layout.count():
-            layout_item = layout.takeAt(0)
-            layout_item.widget().deleteLater()
-
-        # Create the new figure and toolbar widgets. It seems that key
-        # events will not be processed without an mpl figure manager.
-        # However, a figure manager will create a new toplevel window,
-        # which is certainly not desired in this case. This appears to
-        # be a limitation of matplotlib. The canvas is manually set to
-        # visible, or QVBoxLayout will ignore it for size hinting.
+
+        # Create the new canvas and toolbar widgets if suitable ones have
+        # not been created.
+        # The canvas is manually set to visible, or QVBoxLayout will
+        # ignore it for size hinting.
         figure = self.declaration.figure
         if figure:
-            canvas = FigureCanvasQTAgg(figure)
-            canvas.setParent(widget)
-            canvas.setFocusPolicy(Qt.ClickFocus)
-            canvas.setVisible(True)
-            toolbar = NavigationToolbar2QTAgg(canvas, widget)
+            if not self.canvas:
+
+                if isinstance(figure.canvas, FigureCanvasQTAgg):
+                    canvas = figure.canvas
+                else:
+                    canvas = FigureCanvasQTAgg(figure)
+
+                if canvas.toolbar:
+                    # Avoid RuntimeError due to multiple calls to destroy
+                    if hasattr(canvas, 'manager'):
+                        canvas.manager.toolbar = None
+                    toolbar = canvas.toolbar
+                    toolbar.setParent(widget)
+                else:
+                    toolbar = NavigationToolbar2QT(canvas, widget)
+
+                layout.addWidget(toolbar)
+                layout.addWidget(canvas)
+                canvas.setParent(widget)
+                # Use focus policy from MPL FigureManager
+                canvas.setFocusPolicy(Qt.StrongFocus)
+                canvas.setFocus()
```

Comment:
I'm don't think we should be stealing focus here. If the user wants to set focus on the widget, they should call the `set_focus` method on the Enaml widget:
https://github.com/blink1073/enaml/blob/mpl-deprecation/enaml/widgets/widget.py#L191

To make sure the canvas gets focus when that happens, you can reimplement `focus_target` to return the canvas:
https://github.com/blink1073/enaml/blob/mpl-deprecation/enaml/qt/qt_widget.py#L184